### DISCUSSION TITLE
it doesn't seem to work in godot 4.2, but is still an immense help

### DISCUSSION THREAD

**Discussion by jasonswearingen** [Jemy191/GDBridge#4] [General]
I used this to succesfully wrap the phantomcamera gdscript api.  however teh [GlobalClass] attribute seemed to mess up the build, so I copy/pasted the generated code into a new object, removed the GlobalClass attribute and changed the base class to just let me pass in the target Node.

With those changes it is still a ton of help and saved hours.

**Comment by Jemy191** [comment]
Could you give me the code that was messed up? And what change did you made. I will check if I can fix that.

In what way did GlobalClass was messing up the code?
I will remove GlobalClass from the generated file so the user can specify it explicitly.

Also how hard was it to use with PhantomCamera?

**Reply by jasonswearingen** [reply]
It looks like my explanation of the error was slightly wrong.

here's an example, when using this to only gen for the PhantomCamera3d class

```cs
	<ItemGroup>
		<AdditionalFiles Include="**/phantom_camera_3d.gd"/>
	</ItemGroup>
```
I get lots of these errors:

![image](https://github.com/Jemy191/GDBridge/assets/814134/89cc3b1c-87fb-4678-943f-2cc3f3d0a6c7)

it looks like the problem is dupe method names.   the gdscript already has methods named like `set_priority` so the Properties choke because that's the name that the JIT gives the accessor internally.

Also, since I had to manually fix this  I ended up just changing the signature: 

```cs
public class PhantomCamera3DBridgeEz
{
   public const string GDClassName = "PhantomCamera3D";


   protected GodotObject GdObject;

   public PhantomCamera3DBridgeEz(GodotObject gdObject)
   {
      GdObject = gdObject;
      
   }
```
I actually don't know what [GlobalClass or your generated constructor workflow does, but my simple ctor works.


For as for using with PhantomCamera,  I used the `GodotEx` library `[SceneTree]` attribute which helps to generate a reference to gdObjects found in the scene file.   with that and my "simple" binding fix I can just do:

```cs
Node3D pcam = _.PhantomCamera3D;  //a node in my scene
pcamBridge = new PhantomCamera3DBridgeEz(pcam);   //create GDBridge binding
```

all the methods I've used seem to work fine.   But I seem to find some bugs in phantomcamera so I haven't used it much yet.

hope that helps... let me know if you need more info.

**Reply by jasonswearingen** [reply]
oh, sorry I remember now:   when I was using [GlobalClass] I was getting some runtime errors, with it saying the object was already added to the scene tree.  That's why I had to remove it.

**Reply by Jemy191** [reply]
The GDScriptBridge ctor is the same as your PhantomCamera3DBridgeEz ctor.
GlobalClass was so GdScript has completion for C# class.

**Comment by Jemy191** [comment]
Version 0.10.0 should fix all your error. If not, open a Issue so I can check that

**Reply by jasonswearingen** [reply]
yep, 0.10.0 fixes it.  working great now!
